apps/web/screens/ServiceDetails/ServiceDetails.tsx

```diff
@@ -4,27 +4,107 @@ import { withMainLayout } from '@island.is/web/layouts/main'
 import { CustomNextError } from '@island.is/web/units/errors'
 
-const { publicRuntimeConfig } = getConfig()
+import {
+  GetNamespaceQuery,
+  Query,
+  QueryGetApiServiceByIdArgs,
+  QueryGetNamespaceArgs,
+  ApiService,
+} from '@island.is/web/graphql/schema'
+import { GET_NAMESPACE_QUERY, GET_API_SERVICE_QUERY } from '../queries'
+import { SubpageMainContent, ServiceInformation } from '../../components'
+import { SubpageLayout } from '../Layouts/Layouts'
+import SidebarLayout from '../Layouts/SidebarLayout'
+import { Box, LoadingIcon, Text } from '@island.is/island-ui/core'
+import { useNamespace } from '../../hooks'
 
-/* TEMPORARY LAYOUT CREATED TO SCAFFOLD SERVICE DETAILS INTO THE WEB */
+const { publicRuntimeConfig } = getConfig()
 
 interface ServiceDetailsProps {
-  title: string
+  strings: GetNamespaceQuery['getNamespace']
+  loading?: boolean
+  service: ApiService
 }
 
-const ServiceDetails: Screen<ServiceDetailsProps> = ({ title }) => {
+const ServiceDetails: Screen<ServiceDetailsProps> = ({
+  strings,
+  loading = false,
+  service = null,
+}) => {
+  const n = useNamespace(strings)
+
   const { disableApiCatalog: disablePage } = publicRuntimeConfig
 
   if (disablePage === 'true') {
     throw new CustomNextError(404, 'Not found')
   }
 
-  return <h1>{title}</h1>
+  const showService = () => {
+    if (service) {
+      return <ServiceInformation strings={strings} service={service} />
+    }
+
+    return (
+      <Box>
+        <Text variant="h3" as="h3">
+          {n('serviceNotFound')}
+        </Text>
+      </Box>
+    )
+  }
+
+  return (
+    <SubpageLayout
+      main={
+        <SidebarLayout
+          sidebarContent={<>Navigation menu will be displayed here</>}
+        >
+          <SubpageMainContent
+            main={
+              loading ? (
+                <LoadingIcon animate color="blue400" size={32} />
+              ) : (
+                showService()
+              )
+            }
+          />
+        </SidebarLayout>
+      }
+      details={<>The open API document will be displayed here</>}
+    />
+  )
 }
 
 ServiceDetails.getInitialProps = async ({ apolloClient, locale, query }) => {
+  const serviceId = String(query.slug)
+
+  const [filterContent, { data, loading }] = await Promise.all([
+    apolloClient
+      .query<GetNamespaceQuery, QueryGetNamespaceArgs>({
+        query: GET_NAMESPACE_QUERY,
+        variables: {
+          input: {
+            namespace: 'ApiCatalogFilter',
+            lang: locale,
+          },
+        },
+      })
+      .then((res) => JSON.parse(res.data.getNamespace.fields)),
+    apolloClient.query<Query, QueryGetApiServiceByIdArgs>({
+      query: GET_API_SERVICE_QUERY,
+      variables: {
+        input: {
+          id: serviceId,
+        },
+      },
+    }),
+  ])
+
   return {
-    title: 'Vefþjónusta - nánari lýsing',
+    serviceId: serviceId,
+    strings: filterContent,
+    service: data?.getApiServiceById,
+    loading,
```

Comment:
Loading should not be needed unless content changes dynamically and global loading state is not sufficient.
In that case loading should probably be handled inside the cmp since cmp would get the loading state at the same time as the data when using getInitialProps.

Global loading state is handled in router automagically when updating routes.
<img width="1419" alt="loading_bar" src="https://user-images.githubusercontent.com/6640435/102342650-74f5b680-3f91-11eb-9cac-d88f01a54278.png">